utxo/split_dag.go

```diff
@@ -0,0 +1,67 @@
+package utxo
+
+import (
+	"fmt"
+
+	"github.com/xuperchain/xuperunion/pb"
+)
+
+func (uv *UtxoVM) buildTxDeps(txs []*pb.Transaction) (map[string]*pb.Transaction, TxGraph, error) {
+	txMap := map[string]*pb.Transaction{}
+	txGraph := TxGraph{}
+	for _, tx := range txs {
+		txid := fmt.Sprintf("%x", tx.Txid)
+		txMap[txid] = tx
+		txGraph[txid] = []string{}
+	}
+
+	for txid, tx := range txMap {
+		// 填写正常的utxo的引用
+		for _, refTx := range tx.TxInputs {
+			refTxid := string(refTx.RefTxid)
+			// 依赖的交易不在同一个区块中
+			if _, exist := txMap[refTxid]; !exist {
+				continue
+			}
+			txGraph[refTxid] = append(txGraph[refTxid], txid)
+		}
+		// 填写读写集
+		for _, txIn := range tx.TxInputsExt {
+			refTxid := string(txIn.RefTxid)
+			// 依赖的交易不在同一个区块中
+			if _, exist := txMap[refTxid]; !exist {
+				continue
+			}
+			txGraph[refTxid] = append(txGraph[refTxid], txid)
+		}
+	}
+	return txMap, txGraph, nil
+}
+
+// SplitChildDAGs split child dags
+func SplitChildDAGs(txGraph TxGraph, outputTxList []string) []int {
```

Comment:
I suggest migrate this logic of splitting into TopSort, so that can cost lesss computing.